pkg/rpcclient/rpc.go

```diff
@@ -844,234 +836,6 @@ func (c *Client) SubmitRawOracleResponse(ps []any) error {
-		}
-	}
-	txHash = tx.Hash()
-	actualHash, err := c.SendRawTransaction(tx)
-	if err != nil {
-		return txHash, fmt.Errorf("failed to send tx: %w", err)
-	}
-	if !actualHash.Equals(txHash) {
-		return actualHash, fmt.Errorf("sent and actual tx hashes mismatch:\n\tsent: %v\n\tactual: %v", txHash.StringLE(), actualHash.StringLE())
-	}
-	return txHash, nil
-}
-
-// getSigners returns an array of transaction signers and corresponding accounts from
-// given sender and cosigners. If cosigners list already contains sender, the sender
-// will be placed at the start of the list.
-func getSigners(sender *wallet.Account, cosigners []SignerAccount) ([]transaction.Signer, []*wallet.Account, error) {
-	var (
-		signers  []transaction.Signer
-		accounts []*wallet.Account
-	)
-	from := sender.ScriptHash()
-	s := transaction.Signer{
-		Account: from,
-		Scopes:  transaction.None,
-	}
-	for _, c := range cosigners {
-		if c.Signer.Account == from {
-			s = c.Signer
-			continue
-		}
-		signers = append(signers, c.Signer)
-		accounts = append(accounts, c.Account)
-	}
-	signers = append([]transaction.Signer{s}, signers...)
-	accounts = append([]*wallet.Account{sender}, accounts...)
-	return signers, accounts, nil
-}
-
-// SignAndPushP2PNotaryRequest creates and pushes a P2PNotary request constructed from the main
-// and fallback transactions using the given wif to sign it. It returns the request and an error.
-// Fallback transaction is constructed from the given script using the amount of gas specified.
-// For successful fallback transaction validation at least 2*transaction.NotaryServiceFeePerKey
-// GAS should be deposited to the Notary contract.
-// Main transaction should be constructed by the user. Several rules should be met for
-// successful main transaction acceptance:
-//  1. Native Notary contract should be a signer of the main transaction.
-//  2. Notary signer should have None scope.
-//  3. Main transaction should have dummy contract witness for Notary signer.
-//  4. Main transaction should have NotaryAssisted attribute with NKeys specified.
-//  5. NotaryAssisted attribute and dummy Notary witness (as long as the other incomplete witnesses)
-//     should be paid for. Use CalculateNotaryWitness to calculate the amount of network fee to pay
-//     for the attribute and Notary witness.
-//  6. Main transaction either shouldn't have all witnesses attached (in this case none of them
-//     can be multisignature), or it only should have a partial multisignature.
-//
-// Note: client should be initialized before SignAndPushP2PNotaryRequest call.
-//
-// Deprecated: please use Actor from the notary subpackage. This method will be
-// deleted in future versions.
-func (c *Client) SignAndPushP2PNotaryRequest(mainTx *transaction.Transaction, fallbackScript []byte, fallbackSysFee int64, fallbackNetFee int64, fallbackValidFor uint32, acc *wallet.Account) (*payload.P2PNotaryRequest, error) {
```

Comment:
Notary documentation should be adjusted, there's a couple of steps under the following header that should be replaced by Notary Actor usage: https://github.com/nspcc-dev/neo-go/blob/19c59c3a5924bc7e3321e21ce3c961cb76fd5694/docs/notary.md?plain=1#L291

If needed, I can create a separate PR with the documentation update.

Reply:
Fixed.